chain-impl-mockchain/src/vote/payload.rs

```diff
@@ -140,6 +113,67 @@ impl Payload {
     }
 }
 
+impl ProofOfCorrectVote {
+    #[cfg(any(test, feature = "property-test-api"))]
+    pub(crate) fn from_inner(proof: chain_vote::ProofOfCorrectVote) -> Self {
+        assert!(
+            proof.len() <= u8::MAX as usize,
+            "number of options is too large in an internally obtained proof"
+        );
+        Self(proof)
+    }
+
+    pub(super) fn as_inner(&self) -> &chain_vote::ProofOfCorrectVote {
+        &self.0
+    }
+
+    pub(crate) fn serialize_in(&self, bb: ByteBuilder<Self>) -> ByteBuilder<Self> {
+        debug_assert!(self.0.len() <= u8::MAX as usize);
+        bb.u8(self.0.len() as u8)
+            .fold(self.0.ibas(), |bb, iba| bb.bytes(&iba.to_bytes()))
+            .fold(self.0.ds(), |bb, d| bb.bytes(&d.to_bytes()))
+            .fold(self.0.zwvs(), |bb, zwv| bb.bytes(&zwv.to_bytes()))
+            .bytes(&self.0.r().to_bytes())
+    }
+
+    pub fn serialize(&self) -> ByteArray<Self> {
+        self.serialize_in(ByteBuilder::new()).finalize()
+    }
+
+    pub(crate) fn read<'a>(buf: &mut ReadBuf<'a>) -> Result<Self, ReadError> {
```

Comment:
Why `pub(crate)` here?

Reply:
There is no need for this method outside of the crate yet. Can be made public when we need it.